conf/allofus.config

```diff
@@ -0,0 +1,60 @@
+includeConfig 'base.config'
+includeConfig 'container.config'
+
+// Define the number of CPUs based on memory requested
+def calculateCpus(memory) {
+    def mem
+    if (memory instanceof nextflow.util.MemoryUnit) {
+        mem = memory.toMega()
+    } else if (memory instanceof String) {
+        if (memory.toUpperCase().endsWith('G')) {
+            mem = memory[0..-2].toFloat() * 1024
+        } else if (memory.toUpperCase().endsWith('M')) {
+            mem = memory[0..-2].toFloat()
+        } else {
+            mem = memory.toFloat()
+        }
+    } else {
+        mem = memory
+    }
+    return Math.max(1, Math.ceil(mem / 6144).intValue()) // 6144 MB = 6 GB
+}
+
+process {
+    // Override CPUs based on task memory
+    cpus = { calculateCpus(task.memory) }
+
+    withLabel: multithreaded {
+        cpus = { Math.max(2, calculateCpus(task.memory)) }
```

Comment:
This will never be the case for multithreaded mode, but I can see that this is redundant anyways. The only thing is that the number of CPUs must be an even number when > 1. I will need to add a case to handle this for the multithreaded label.